motionplan/motionPlanner.go

```diff
@@ -457,30 +457,30 @@ func CheckPlan(
 	currentInputs []frame.Input,
 	errorState spatialmath.Pose,
 	logger *zap.SugaredLogger,
-) error {
+) (spatialmath.Pose, error) {
 	// ensure that we can actually perform the check
 	if len(plan) < 1 {
-		return errors.New("plan must have at least one element")
+		return nil, errors.New("plan must have at least one element")
 	}
 
 	// construct solverFrame
 	// Note that this requires all frames which move as part of the plan, to have an
 	// entry in the very first plan waypoint
 	sf, err := newSolverFrame(fs, checkFrame.Name(), frame.World, plan[0])
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	// construct planager
+	// construct planaer
```

Comment:
[nit] Either leave this comment as is or change to "plan manager"

Reply:
I'm a fan of planager :)